pkg/elevate/elevate.go

```diff
@@ -73,42 +72,24 @@ func RunElevate(argv []string) error {
 		return err
 	}
 
-	logger.Debug("Adding impersonation RBAC allow permissions to kubeconfig")
-
-	elevateCmd := "oc " + strings.Join(argv[1:], " ")
-
-	// Make the Default SHELL environment variable to /bin/bash
-	shell := os.Getenv("SHELL")
-	if shell == "" || !utils.ShellChecker.IsValidShell(shell) {
-		if utils.ShellChecker.IsValidShell("/bin/bash") {
-			shell = "/bin/bash"
-		} else {
-			return fmt.Errorf("both the SHELL environment variable and /bin/bash are not set or invalid. Please ensure a valid shell is set in your environment")
+	// As WriteKubeconfigToFile is also creating a tempory file reference by new KUBECONFIG variable setting,
+	// we need to take care of its cleanup
+	tempKubeconfigPath, _ := os.LookupEnv("KUBECONFIG")
+	defer func() {
+		logger.Debugln("Cleaning up temporary kubeconfig", tempKubeconfigPath)
+		err := OsRemove(tempKubeconfigPath)
+		if err != nil {
+			fmt.Println(err)
 		}
-	}
+	}()
 
 	logger.Debugln("Executing command with temporary kubeconfig as backplane-cluster-admin")
-
-	ocCmd := ExecCmd(shell, "-c", elevateCmd)
```

Comment:
Wondering why it uses `bash -c` at the first place, have you confirmed the change works on both linux and mac?

Reply:
The reason of using bash -c was for [OSD-18058](https://issues.redhat.com/browse/OSD-18058), but as mentioned in [OSD-22522](https://issues.redhat.com/browse/OSD-22522) comment, this usage is not working as it was expecting and it is not needed as there is no need for it to have interactive prompt.